mne/preprocessing/otp.py

```diff
@@ -0,0 +1,112 @@
+# -*- coding: utf-8 -*-
+# Authors: Samu Taulu <staulu@uw.edu>
+#          Eric Larson <larson.eric.d@gmail.com>
+
+# License: BSD (3-clause)
+
+import numpy as np
+from scipy import linalg
+
+from ..cov import _check_scalings_user, _apply_scaling_array
+from ..io.pick import pick_info, _picks_by_type, _pick_data_channels
+from ..utils import logger, verbose
+
+
+@verbose
+def oversampled_temporal_projection(raw, duration=10., scalings=None,
+                                    verbose=None):
+    """Denoise MEG channels using oversampled temporal projection
+
+    This algorithm uses a leave-one-out procedure with parallel
+    temporal projection to remove individual sensor noise under the
+    assumption that sampled fields (e.g., MEG and EEG) are oversampled
+    by the sensor array [1]_.
+
+    .. note:: Bad channels are not used or modified by this procedure.
+
+    Parameters
+    ----------
+    raw : instance of Raw
+        Raw data to denoise.
+    duration : float
+        The window duration (in seconds) to use (default: 10.).
+    scalings : dict | None
+        If None (default), use ``dict(mag=1e15, grad=1e13, eeg=1e6)``.
+        These defaults will scale different channel types to approximately
+        at the same unit.
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see mne.verbose)
+
+    Returns
+    -------
+    raw_clean : instance of Raw
+        The cleaned data.
+
+    Notes
+    -----
+    This algorithm is computationally expensive, and can be several times
+    slower than realtime for conventional M/EEG datasets.
+
+    This algorithm is covered by a provisional patent.
+
+    .. versionadded:: 0.14
+
+    References
+    ----------
+    .. [1] Taulu, S and Larson E (2016). "Reducing noise in
+           electromagnetic sensor arrays using oversampled temporal
+           projection", presented at the 2016 International Conference
+           on Biomagnetism, Seoul, Korea 3 Oct 2016.
+    """
+    # XXX eventually offer option to treat "bads" by projecting them onto
+    # others, but not use them in bases for others?
+    logger.info('Processing MEG data using oversampled temporal projection')
+    scalings = _check_scalings_user(scalings)
+    n_samp = int(round(float(duration) * raw.info['sfreq']))
```

Comment:
Well this is multiplication... but I guess here we should be guaranteed that `raw.info['sfreq']` is a float anyway, so you're right this is overkill